### PR TITLE
More stream tests

### PR DESCRIPTION
**Pull request by dahliamalkhi**
This small PR introduces a minor boundary-fix to backpointerStreamView.java, which yields significant performance improvement (on my machine).

I also include a few minor test cleanups and another stream test.

### CODE REVIEW

File: test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java [hunk 1]
```diff
@@ -19,17 +19,78 @@
...
         assertThat(sv.hasNext()).isFalse();
     }
+
+    @Test
+    public void readQueueTest() {
+        CorfuRuntime runtime = getDefaultRuntime();
+        IStreamView sv = runtime.getStreamsView().get(CorfuRuntime.getStreamID("streamA"));
+        final int ten = 10;
+
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_LOW; i++)
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java]: Add braces to the loop's body.

File: test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java [hunk 2]
```diff
@@ -19,17 +19,78 @@
...
+    }
+
+
+    @Test
+    public void moreReadQueueTest() {
+        CorfuRuntime runtime = getDefaultRuntime();
+        IStreamView sv = runtime.getStreamsView().get(CorfuRuntime.getStreamID("streamA"));
+        final int ten = 10;
+
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_VERY_LOW; i++)
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java]: Add braces.

File: test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java [hunk 3]
```diff
@@ -19,17 +19,78 @@
...
         IStreamView sv = runtime.getStreamsView().get(CorfuRuntime.getStreamID("streamA"));
         sv.append("hello world".getBytes());
 
         assertThat(sv.hasNext()).isTrue();
         sv.next();
         assertThat(sv.hasNext()).isFalse();
     }
+
+    @Test
+    public void readQueueTest() {
```
Comment (medhavidhawan) [inline test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java]: comments would be good.

**Comment by dahliamalkhi** [comment]
hmm.. good and short code doesn't need comments (really, this test doesn't), and good C programmers doen't put braces around single line statements. Really. I think we should merge, the test is less important than the optimization to backpointersStreamview.

File: test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java [hunk 4]
```diff
@@ -19,17 +19,78 @@
...
+            sv.append(String.valueOf(i).getBytes());
+
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_LOW; i++) {
+            assertThat(sv.hasNext()).isTrue();
+            byte[] payLoad = (byte[]) sv.next().getPayload(runtime);
+            assertThat(new String(payLoad).equals(String.valueOf(i)))
+                    .isTrue();
+            assertThat(sv.getCurrentGlobalPosition()).isEqualTo(i);
+
+            if (i % ten == 1) {
```
Comment (medhavidhawan) [inline test/src/test/java/org/corfudb/runtime/view/stream/BackpointerStreamViewTest.java]: it is hard to understand what this is testing unless we put in some comments

**Comment by dahliamalkhi** [comment]
OK, OK. Addressed reviewers' style comments. Thx @Maithem @medhavidhawan

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **0af4041**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #665 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/A7a5euNUEawYcm8nbi7YfnwYz2L5oXDE)

**Pull request merged**